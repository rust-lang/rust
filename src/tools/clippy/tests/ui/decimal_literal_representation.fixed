#[warn(clippy::decimal_literal_representation)]
#[allow(unused_variables)]
#[rustfmt::skip]
fn main() {
    let good = (       // Hex:
        127,           // 0x7F
        256,           // 0x100
        511,           // 0x1FF
        2048,          // 0x800
        4090,          // 0xFFA
        16_371,        // 0x3FF3
        61_683,        // 0xF0F3
        2_131_750_925, // 0x7F0F_F00D
    );
    let bad = (        // Hex:
        0x8005,        // 0x8005
        0xFF00,        // 0xFF00
        0x7F0F_F00F, // 0x7F0F_F00F
        0x7FFF_FFFF, // 0x7FFF_FFFF
        #[allow(overflowing_literals)]
        0xF0F0_F0F0, // 0xF0F0_F0F0
        0x8005_usize,   // 0x8005_usize
        0x7F0F_F00F_isize, // 0x7F0F_F00F_isize
    );
}
