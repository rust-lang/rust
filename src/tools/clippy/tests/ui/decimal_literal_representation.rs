#[warn(clippy::decimal_literal_representation)]
#[allow(unused_variables)]
#[rustfmt::skip]
fn main() {
    let good = (       // Hex:
        127,           // 0x7F
        256,           // 0x100
        511,           // 0x1FF
        2048,          // 0x800
        4090,          // 0xFFA
        16_371,        // 0x3FF3
        61_683,        // 0xF0F3
        2_131_750_925, // 0x7F0F_F00D
    );
    let bad = (        // Hex:
        32_773,        // 0x8005
        65_280,        // 0xFF00
        2_131_750_927, // 0x7F0F_F00F
        2_147_483_647, // 0x7FFF_FFFF
        #[allow(overflowing_literals)]
        4_042_322_160, // 0xF0F0_F0F0
        32_773usize,   // 0x8005_usize
        2_131_750_927isize, // 0x7F0F_F00F_isize
    );
}
