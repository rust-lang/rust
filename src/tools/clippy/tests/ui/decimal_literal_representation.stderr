error: integer literal has a better hexadecimal representation
  --> tests/ui/decimal_literal_representation.rs:16:9
   |
LL |         32_773,        // 0x8005
   |         ^^^^^^ help: consider: `0x8005`
   |
   = note: `-D clippy::decimal-literal-representation` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::decimal_literal_representation)]`

error: integer literal has a better hexadecimal representation
  --> tests/ui/decimal_literal_representation.rs:17:9
   |
LL |         65_280,        // 0xFF00
   |         ^^^^^^ help: consider: `0xFF00`

error: integer literal has a better hexadecimal representation
  --> tests/ui/decimal_literal_representation.rs:18:9
   |
LL |         2_131_750_927, // 0x7F0F_F00F
   |         ^^^^^^^^^^^^^ help: consider: `0x7F0F_F00F`

error: integer literal has a better hexadecimal representation
  --> tests/ui/decimal_literal_representation.rs:19:9
   |
LL |         2_147_483_647, // 0x7FFF_FFFF
   |         ^^^^^^^^^^^^^ help: consider: `0x7FFF_FFFF`

error: integer literal has a better hexadecimal representation
  --> tests/ui/decimal_literal_representation.rs:21:9
   |
LL |         4_042_322_160, // 0xF0F0_F0F0
   |         ^^^^^^^^^^^^^ help: consider: `0xF0F0_F0F0`

error: integer literal has a better hexadecimal representation
  --> tests/ui/decimal_literal_representation.rs:22:9
   |
LL |         32_773usize,   // 0x8005_usize
   |         ^^^^^^^^^^^ help: consider: `0x8005_usize`

error: integer literal has a better hexadecimal representation
  --> tests/ui/decimal_literal_representation.rs:23:9
   |
LL |         2_131_750_927isize, // 0x7F0F_F00F_isize
   |         ^^^^^^^^^^^^^^^^^^ help: consider: `0x7F0F_F00F_isize`

error: aborting due to 7 previous errors

